File: Java/Medium/Accepted/sort-colors_9.java
class Solution {
    public void sortColors(int[] nums) {
        
        int p0 = 0, p2 = nums.length - 1, cur = 0;
        while (cur <= p2) {
            if (nums[cur] == 0) {
                int tmp = nums[cur];
                nums[cur] = nums[p0];
                nums[p0] = tmp;
                p0++;
                cur++;
            } else if (nums[cur] == 2) {
                int tmp = nums[cur];
                nums[cur] = nums[p2];
                nums[p2] = tmp;
                p2--;
            } else {
                cur++;
            }
        }
    }
}

File: C++/Medium/Accepted/sort-colors_9.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        
        int count[3] = {0};
        for (auto i : nums) {
            count[i]++;
        }
        int j = 0;
        for (int i = 0; i < 3; i++) {
            while (count[i]-- > 0) {
                nums[j++] = i;
            }
        }
    }
};